service/src/test/java/uk/gov/hmcts/reform/fpl/service/NoticeOfProceedingsTemplateDataGenerationServiceTest.java

```diff
@@ -0,0 +1,238 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.ChildParty;
+import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+import uk.gov.hmcts.reform.fpl.model.NoticeOfProceedings;
+import uk.gov.hmcts.reform.fpl.model.Orders;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.JudgeAndLegalAdvisor;
+import uk.gov.hmcts.reform.fpl.model.docmosis.DocmosisNoticeOfProceeding;
+import uk.gov.hmcts.reform.fpl.service.config.LookupTestConfig;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+import uk.gov.hmcts.reform.fpl.utils.FixedTimeConfiguration;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.FormatStyle;
+import java.util.List;
+import java.util.UUID;
+
+import static java.util.Collections.emptyList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.enums.JudgeOrMagistrateTitle.HIS_HONOUR_JUDGE;
+import static uk.gov.hmcts.reform.fpl.enums.JudgeOrMagistrateTitle.MAGISTRATES;
+import static uk.gov.hmcts.reform.fpl.enums.OrderType.CARE_ORDER;
+import static uk.gov.hmcts.reform.fpl.enums.OrderType.EDUCATION_SUPERVISION_ORDER;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createHearingBooking;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createPopulatedApplicants;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createPopulatedChildren;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.formatLocalDateToString;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {NoticeOfProceedingsTemplateDataGenerationService.class})
+@ContextConfiguration(classes = {
+    JacksonAutoConfiguration.class, LookupTestConfig.class, HearingBookingService.class,
+    HearingVenueLookUpService.class, CaseDataExtractionService.class, FixedTimeConfiguration.class
+})
+class NoticeOfProceedingsTemplateDataGenerationServiceTest {
+
+    @Autowired
+    private Time time;
+
+    @Autowired
+    private NoticeOfProceedingsTemplateDataGenerationService noticeOfProceedingsTemplateDataGenerationService;
+
+    private LocalDate futureDate;
+    private List<Element<Child>> children;
+
+    @BeforeEach
+    void setup() {
+        futureDate = time.now().toLocalDate().plusDays(1);
+        children = createPopulatedChildren(time.now().toLocalDate());
```

Comment:
we can have `createPopulatedChildren(time.now().toLocalDate())` added to `initNoticeOfProceedingCaseData()`. This is because its used everywhere else expect from `shouldNotApplySentenceFormattingWhenOnlyOneChildExistsOnCase()` which builds its own

Reply:
agree, Done. Good job for looking at this :thumbsup:.